core/drivers/crypto/stm32/stm32_pka.c

```diff
@@ -0,0 +1,1740 @@
+				0x00
+			},
+			.size = U(65),
+		},
+		.g = {
+			.x = {
+				.val = (uint8_t[]){
+					0xc6, 0x85, 0x8e, 0x06,
+					0xb7, 0x04, 0x04, 0xe9,
+					0xcd, 0x9e, 0x3e, 0xcb,
+					0x66, 0x23, 0x95, 0xb4,
+					0x42, 0x9c, 0x64, 0x81,
+					0x39, 0x05, 0x3f, 0xb5,
+					0x21, 0xf8, 0x28, 0xaf,
+					0x60, 0x6b, 0x4d, 0x3d,
+					0xba, 0xa1, 0x4b, 0x5e,
+					0x77, 0xef, 0xe7, 0x59,
+					0x28, 0xfe, 0x1d, 0xc1,
+					0x27, 0xa2, 0xff, 0xa8,
+					0xde, 0x33, 0x48, 0xb3,
+					0xc1, 0x85, 0x6a, 0x42,
+					0x9b, 0xf9, 0x7e, 0x7e,
+					0x31, 0xc2, 0xe5, 0xbd,
+					0x66
+				},
+				.size = U(65),
+			},
+			.y = {
+				.val = (uint8_t[]){
+					0x01, 0x18, 0x39, 0x29,
+					0x6a, 0x78, 0x9a, 0x3b,
+					0xc0, 0x04, 0x5c, 0x8a,
+					0x5f, 0xb4, 0x2c, 0x7d,
+					0x1b, 0xd9, 0x98, 0xf5,
+					0x44, 0x49, 0x57, 0x9b,
+					0x44, 0x68, 0x17, 0xaf,
+					0xbd, 0x17, 0x27, 0x3e,
+					0x66, 0x2c, 0x97, 0xee,
+					0x72, 0x99, 0x5e, 0xf4,
+					0x26, 0x40, 0xc5, 0x50,
+					0xb9, 0x01, 0x3f, 0xad,
+					0x07, 0x61, 0x35, 0x3c,
+					0x70, 0x86, 0xa2, 0x72,
+					0xc2, 0x40, 0x88, 0xbe,
+					0x94, 0x76, 0x9f, 0xd1,
+					0x66, 0x50
+				},
+				.size = U(66),
+			},
+		},
+	},
+};
+
+struct stm32_pka_platdata {
+	vaddr_t base;
+	struct clk *clk;
+	struct clk *clk_rng;
+	struct rstctrl *reset;
+	/* Protect PKA HW instance access */
+	struct mutex *lock;
+};
+
+static struct stm32_pka_platdata pka_pdata;
+static struct mutex pka_lock = MUTEX_INITIALIZER;
+
+static TEE_Result pka_wait_bit(const vaddr_t base, const uint32_t bit_mask)
+{
+	uint32_t value = 0;
+
+	if (IO_READ32_POLL_TIMEOUT(base + _PKA_SR, value,
+				   (value & bit_mask) == bit_mask, 0,
+				   PKA_TIMEOUT_US)) {
+		DMSG("timeout waiting 0x%"PRIx32, bit_mask);
+		return TEE_ERROR_BUSY;
+	}
+
+	return TEE_SUCCESS;
+}
+
+static void pka_disable(const vaddr_t base)
+{
+	io_clrbits32(base + _PKA_CR, _PKA_CR_EN);
+}
+
+static TEE_Result pka_enable(const vaddr_t base, const uint32_t mode)
+{
+	/* Set mode and disable interrupts */
+	io_clrsetbits32(base + _PKA_CR, _PKA_IT_MASK | _PKA_CR_MODE_MASK,
+			SHIFT_U32(mode, _PKA_CR_MODE_SHIFT));
+
+	io_setbits32(base + _PKA_CR, _PKA_CR_EN);
+
+	return pka_wait_bit(base, _PKA_SR_INITOK);
+}
+
+/*
+ * Data are already loaded in PKA internal RAM,
+ * MODE is set,
+ * we start process, and wait for its end.
+ */
+static TEE_Result stm32_pka_process(const vaddr_t base)
+{
+	io_setbits32(base + _PKA_CR, _PKA_CR_START);
+
+	return pka_wait_bit(base, _PKA_IT_PROCEND);
+}
+
+/**
+ * Read ECC operand from PKA RAM
+ *
+ * PKA reads u64 words, for each u64 LSB is bit 0, MSB is bit 63.
+ * We read @eo_nbw (ECC operand Size) u64. The value of @eo_nbw depends
+ * on the chosen prime modulus length in bits.
+ *
+ * First less significant u64 is read from lowest address.
+ * Most significant u64 from higher address.
+ * Last u64 is expect to be equal to 0x0.
+ *
+ * This function manage endianness (as bswap64 do) and padding incomplete u64
```

Comment:
s/manage/manages/
s/padding/padding of/   or   s/padding/pads/

Below: s/0/0s/ (ditto at line 760)